File: leetcode/contains-duplicate-ii.java
// https://leetcode.com/submissions/detail/786783940/

import java.util.HashMap;

class Solution {
    public boolean containsNearbyDuplicate(int[] nums, int k) {
        HashMap<Integer, Integer> map = new HashMap<Integer, Integer>();
        
        for(int i = 0; i<nums.length; i++){
            if(map.get(nums[i]) != null && (i - map.get(nums[i])) <= k) return true;
            map.put(nums[i], i);
        }
        return false;
    }
}

File: leetcode/contains_duplicate_ii.py
# https://leetcode.com/problems/contains-duplicate-ii


class Solution:
    def containsNearbyDuplicate(self, nums: list[int], k: int) -> bool:
        range_dict: dict[int] = {}
        left: int = 0
        right: int = 0

        while right < len(nums) and right <= k:
            if range_dict.get(nums[right]):
                return True
            range_dict[nums[right]] = 1
            right += 1

        while right < len(nums):
            range_dict[nums[left]] = None
            left += 1

            if range_dict.get(nums[right]):
                return True

            range_dict[nums[right]] = 1
            right += 1

        return False
